crates/modelardb_common/src/storage.rs

```diff
@@ -330,6 +335,34 @@ pub async fn extract_azure_blob_storage_arguments(
     Ok((account, access_key, container_name, offset_data))
 }
 
+/// Reinterpret the bits used for univariate ids in `compressed_segments` to convert the column from
+/// [`UInt64Array`] to [`Int64Array`] as the Delta Lake Protocol does not support unsigned integers.
+fn univariate_ids_uint64_to_int64(compressed_segments: &mut Vec<RecordBatch>) {
+    for record_batch in compressed_segments {
+        let mut columns = record_batch.columns().to_vec();
+        let univariate_ids = crate::array!(record_batch, 0, UInt64Array);
+        let signed_univariate_ids: Int64Array =
+            univariate_ids.unary(|value| i64::from_ne_bytes(value.to_ne_bytes()));
+        columns[0] = Arc::new(signed_univariate_ids);
+
+        // unwrap() is safe as columns are constructs to match DISK_COMPRESSED_SCHEMA.
+        *record_batch = RecordBatch::try_new(DISK_COMPRESSED_SCHEMA.0.clone(), columns).unwrap();
+    }
+}
+
+/// Reinterpret the bits used for univariate ids in `compressed_segments` to convert the column from
```

Comment:
Maybe make it clear that a new record batch is returned to avoid confusion with the above function that looks similar but instead converts the given record batch directly.

Reply:
Fixed.